project-base/app/tests/App/Functional/Form/AutoCardGroupingRenderingTest.php

```diff
@@ -0,0 +1,307 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Tests\App\Functional\Form;
+
+use DOMDocument;
+use DOMNodeList;
+use DOMXPath;
+use Override;
+use Shopsys\FrameworkBundle\Form\GroupType;
+use Symfony\Component\Form\Extension\Core\Type\HiddenType;
+use Symfony\Component\Form\Extension\Core\Type\TextType;
+use Symfony\Component\Form\FormBuilderInterface;
+use Symfony\Component\Form\FormFactoryInterface;
+use Symfony\Component\Form\FormInterface;
+use Tests\App\Test\FunctionalTestCase;
+use Twig\Environment;
+
+class AutoCardGroupingRenderingTest extends FunctionalTestCase
+{
+    private const string XPATH_CARD = '//div[contains(@class, "card") and contains(@class, "mb-3")]';
+    private const string XPATH_CARD_BODY = '//div[contains(@class, "card-body")]';
+    private const string XPATH_CARD_BODY_ROWS = '//div[contains(@class, "card-body")]//div[contains(@class, "row")]';
```

Comment:
maybe we can concatenate `XPATH_CARD_BODY` constant here to avoid the duplication of `div[contains(@class, "card-body")]` xpath definition?